internal/k8s/reconciler/route.go

```diff
@@ -411,8 +415,10 @@ func (r *K8sRoute) Validate(ctx context.Context) error {
 		if err != nil {
 			return err
 		} else if !allowed {
-			// FUTURE Set appropriate status and early out
-			r.logger.Warn("Cross-namespace routing not allowed, will enforce in future", "refName", ref.Name, "refNamespace", ref.Namespace)
+			msg := fmt.Sprintf("Cross-namespace routing not allowed without matching referencepolicy refName: %v refNamespace: %v", ref.Name, *ref.Namespace)
+			err := service.NewRefNotPermittedError(msg)
+			r.resolutionErrors.Add(err)
```

Comment:
same as above, no real need for the intermediate assignment and the line isn't that long if you just inline the error construction:

```suggestion
			r.resolutionErrors.Add(service.NewRefNotPermittedError(msg))
```